src/mlpack/methods/gmm/em_fit_impl.hpp

```diff
@@ -33,23 +35,20 @@ EMFit<InitialClusteringType, CovarianceConstraintPolicy>::EMFit(
     constraint(constraint)
 { /* Nothing to do. */ }
 
-template<typename InitialClusteringType, typename CovarianceConstraintPolicy>
-void EMFit<InitialClusteringType, CovarianceConstraintPolicy>::Estimate(
-    const arma::mat& observations,
-    std::vector<distribution::GaussianDistribution>& dists,
-    arma::vec& weights,
-    const bool useInitialModel)
+template<typename InitialClusteringType,
+         typename CovarianceConstraintPolicy,
+         typename Distribution>
+void EMFit<InitialClusteringType, CovarianceConstraintPolicy, Distribution>::
+Estimate(const arma::mat& observations,
+         std::vector<Distribution>& dists,
+         arma::vec& weights,
+         const bool useInitialModel)
 {
-  // Shortcut: if the user is using the DiagonalConstraint, then we will call
-  // out to Armadillo.  But Armadillo uses uword internally as an OpenMP index
-  // type, which crashes Visual Studio, so don't do this on Windows.
-  #ifndef _WIN32
   if (std::is_same<CovarianceConstraintPolicy, DiagonalConstraint>::value)
   {
     ArmadilloGMMWrapper(observations, dists, weights, useInitialModel);
     return;
   }
```

Comment:
To expand on my earlier thought, here, I think we could do this:

```
if (std::is_same<Distribution, DiagonalGaussianDistribution>::value)
{
  // or _MSVC?
  #ifdef _WIN32 and ARMA_USE_OPENMP
    Log::Warn << "Cannot use arma::gmm_diag on Visual Studio!" ...
  #else 
    ArmadilloGMMWrapper(observations, dists, weights, useInitialModel);
    return;
  #endif 
}
```

The fallback here will just use the rest of this function.

Reply:
Agreed :+1